packages/extension-ui/src/Popup/AuthManagement/DisconnectApp.tsx

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ThemeProps } from '../../types';
+
+import React, { useCallback, useContext } from 'react';
+import { useParams } from 'react-router';
+import styled from 'styled-components';
+
+import animatedRemove from '../../assets/anim_unlink.svg';
+import { Button, ButtonArea, FaviconBox, Svg, VerticalSpace } from '../../components';
+import { ActionContext } from '../../components/contexts';
+import useToast from '../../hooks/useToast';
+import useTranslation from '../../hooks/useTranslation';
+import { removeAuthorization } from '../../messaging';
+import Header from '../../partials/Header';
+
+interface Props extends ThemeProps {
+  className?: string;
+}
+interface URLState {
+  url: string;
+}
+
+const CustomFaviconBox = styled(FaviconBox)`
+    width: 270px;
+    box-sizing: border-box;
+    margin: 0 auto;
+    margin-top: 16px;
+`;
+
+function DisconnectApp({ className }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const { url } = useParams<URLState>();
+  const decodedUrl = decodeURIComponent(url);
+  const onAction = useContext(ActionContext);
+  const { show } = useToast();
+
+  const goTo = useCallback((path: string) => () => onAction(path), [onAction]);
+
+  const removeAuth = useCallback(
+    (url: string) => {
+      show(t<string>('App disconnected'), 'success', () => {
+        removeAuthorization(url)
+          .then(() => onAction('/auth-list'))
+          .catch(console.error);
+      });
+      goTo('/auth-list');
+    },
+    [goTo, onAction, show, t]
+  );
+
+  const _removeAuth = useCallback(() => removeAuth(decodeURIComponent(url)), [removeAuth, url]);
```

Comment:
Why can't we fire removeAuth directly but have this and removeAuth in useCallback?